tests/subsys/bluetooth/mesh/light_hue/src/main.c

```diff
@@ -0,0 +1,688 @@
+		uint16_t max;
+		uint16_t current_hue;
+		uint16_t target_hue;
+		uint16_t expected_hue;
+	} test_vector[] = {
+		/* Range Min, Range Max, Current, Target, Expected */
+		{  0,         0xFFFF,    0,       0,      0       },
+		{  0,         0xFFFF,    0,       0xFFFF, 0xFFFF  },
+		/* [100, 0xFF00] */
+		{  100,       0xFF00,    0xFFFF,  99,     100     },
+		{  100,       0xFF00,    100,     0xFF01, 0xFF00  },
+		{  100,       0xFF00,    0xFF00,  101,    101     },
+		{  100,       0xFF00,    101,     0xFEFF, 0xFEFF  },
+		/* [0 - 100, 0xFF00 - 0xFFFF] */
+		{  0xFF00,    100,       0xFEFF,  101,    100     },
+		{  0xFF00,    100,       100,     0xFEFF, 0xFF00  },
+		{  0xFF00,    100,       0xFF00,  99,     99      },
+		{  0xFF00,    100,       99,      0xFF01, 0xFF01  },
+	};
+
+	for (size_t i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		struct bt_mesh_light_hue_status status = {};
+
+		printk("Checking test_vector[%d]\n", i);
+
+		light_hue_srv.range.min = test_vector[i].min;
+		light_hue_srv.range.max = test_vector[i].max;
+		light_hue_val = test_vector[i].current_hue;
+
+		status.current = test_vector[i].expected_hue;
+		expect_hue_status_pub(&status);
+
+		struct bt_mesh_light_hue hue_set = {
+			.lvl = test_vector[i].target_hue,
+		};
+		struct bt_mesh_light_hue exp_hue_set = {
+			.lvl = test_vector[i].expected_hue,
+		};
+
+		expect_hue_set_cb(&exp_hue_set);
+		zassert_ok(hue_set_unack_send(&hue_set));
+	}
+}
+
+/* Test Hue value clamping with transition. */
+ZTEST(light_hue_srv_test, test_hue_set_transition)
+{
+	struct {
+		uint16_t min;
+		uint16_t max;
+		uint16_t current_hue;
+		uint16_t target_hue;
+		uint16_t expected_hue;
+		int direction;
+	} test_vector[] = {
+		/* Range Min, Range Max, Current, Target, Expected, Diretion */
+		{  0,         0xFFFF,    0,       0,      0,         1  },
+		{  0,         0xFFFF,    0,       0xFFFF, 0xFFFF,   -1  },
+		/* Direction points to the shortest path. */
+		{  0,         0xFFFF,    0xFFFF,  0,      0,         1  },
+		{  0,         0xFFFF,    0,       0xFFFF, 0xFFFF,   -1  },
+		/* [100, 0xFF00] */
+		{  100,       0xFF00,    0xFFFF,  99,     100,      -1  },
+		{  100,       0xFF00,    100,     0xFF01, 0xFF00,    1  },
+		{  100,       0xFF00,    0xFF00,  101,    101,      -1  },
+		{  100,       0xFF00,    101,     0xFEFF, 0xFEFF,    1  },
+		/* [0 - 100, 0xFF00 - 0xFFFF] */
+		{  0xFF00,    100,       0xFEFF,  101,    100,       1  },
+		{  0xFF00,    100,       100,     0xFEFF, 0xFF00,   -1  },
+		{  0xFF00,    100,       0xFF00,  99,     99,        1  },
+		{  0xFF00,    100,       99,      0xFF01, 0xFF01,   -1  },
+	};
+
+	for (size_t i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		struct bt_mesh_light_hue_status status = {
+			/* Test stores previous value and returns it as current in the next
+			 * iteration.
+			 */
+			.current = test_vector[i].current_hue,
+			.target = test_vector[i].expected_hue,
+			.remaining_time = 100, /* Same as Transition Time set below. */
+		};
+
```

Comment:
If this is true:
```c
/* Test stores previous value and returns it as current in the next
 * iteration.
 */
```
then perhaps you could remove the explicit `current_hue` from the vector and do this?

```suggestion
	uint16_t current_hue = 0;

	for (size_t i = 0; i < ARRAY_SIZE(test_vector); i++) {
		struct bt_mesh_light_hue_status status = {
			/* Test stores previous value and returns it as current in the next
			 * iteration.
			 */
			.current = current_hue,
			.target = test_vector[i].expected_hue,
			.remaining_time = 100, /* Same as Transition Time set below. */
		};

		current_hue = test_vector[i].expected_hue;
```

to remove the possibility of creating "obviously invalid" test vectors? This is just a suggestion, either way is fine by me. If you change it then do the same everywhere. 

After thinking more about it, I see the value of being able to read the "current" easily when reading the table. Up to you what to do here.

Reply:
Ah, it is an old comment which I forgot to remove.